crates/pubsub/src/implementors/mosquitto.rs

```diff
@@ -65,31 +86,93 @@ impl MosquittoImplementor {
     }
 }
 
-// Sub
-impl MosquittoImplementor {
-    pub async fn subscribe(&self, topic: &str) -> Result<()> {
-        block_in_place(|| {
+impl Sub {
+    pub async fn new(slight_state: &BasicState) -> Self {
+        let host = get_from_state("MOSQUITTO_HOST", slight_state)
+            .await
+            .unwrap();
+        let port = get_from_state("MOSQUITTO_PORT", slight_state)
+            .await
+            .unwrap()
+            .parse::<i32>()
+            .unwrap();
+
+        tracing::info!("Connecting to Mosquitto broker at {}:{}", host, port);
+
+        // let ( consumer, subscriptions ) = block_in_place(|| {
+        //     Handle::current().block_on(async move {
+        //         let mut client = Client::with_auto_id().unwrap();
+
+        //         client
+        //             .connect(&host, port, std::time::Duration::from_secs(5), None)
+        //             .await
+        //             .unwrap();
+
+        //         let ret0 = Arc::new(Mutex::new(client));
+        //         let ret1 = Arc::new(Mutex::new(None));
+
+        //         (ret0, ret1)
+        //     })
+        // });
+
+        Self {
+            host,
+            port,
+            consumers: Arc::new(Mutex::new(HashMap::new())),
+        }
+    }
+
+    pub async fn subscribe(&self, topic: &str) -> Result<String> {
+        let new_consumer = block_in_place(|| {
             Handle::current().block_on(async move {
-                self.client
-                    .lock()
-                    .unwrap()
-                    .subscribe(topic, QoS::AtMostOnce)
+                let mut client = Client::with_auto_id().unwrap();
+
+                client
```

Comment:
I think this implies that each time subscribe is called, then a new TCP connection will be made to the broker. What do you think about creating a client per named capability, rather than per subscription? I could imagine a lot of TCP connections if it is done per subscription.

Reply:
I see two problems w/ that.

(1) Due to the fact that, in several implementors, `subscribe`s are tied to a client instance — We need a new client per subscription to be able to differentiate them and get a proper subscription token that guarantees a `receive` for that one topic. 

(2) In addition, capabilities are established at compilation time — If we follow tie subscriptions to named capabilities, we'd have to know how many subscriptions we want then, which would restrict our coverage of user applications.

Reply:
Ahh... I see. Thank you for pointing that out. I was looking at the mosquito implementation. After looking at the Kafka implementation your point becomes clear.

Reply:
I tried looking to see if I could change the Mosquitto implementation, but have had no luck so far. With that, is this PR good to merge for you? After this, I'm gonna move the stuff that implements `mq` to implement `pubsub` instead and complete the to `messaging` re-name  — I'm hoping that I'll get some inspiration when doing the service bus implementation 😆

Reply:
Thank you for giving it a try. We'll probably have to come back to it at some point in the future, but let's not impede your work.